src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -1392,37 +1415,130 @@ export function intervalFiltersDisplayValue(values: DataFilterValue[]) {
+                setValue={val => {
+                    if (!val) {
+                        // empty end
+                        onUpdate({ start });
+                    } else {
+                        const valNum = parseFloat(val);
+                        if (valNum < start!) {
+                            // invert range
+                            onUpdate({ start: valNum, end: start });
+                        } else {
+                            onUpdate({ start, end: valNum });
+                        }
+                    }
+                }}
+                numericOnly={true}
+                allowEmptyValue={true}
+                textFieldAppearance={true}
+            />
+        );
+        const startEqualsEndText = stringOutput ? (
+            startValue
+        ) : (
+            <EditableSpan
+                value={startValue}
+                setValue={val => {
+                    if (!val) {
+                        // empty
+                        onUpdate({ start: undefined, end: undefined });
+                    } else {
+                        onUpdate({
+                            start: parseFloat(val),
+                            end: parseFloat(val),
+                        });
+                    }
+                }}
+                numericOnly={true}
+                allowEmptyValue={true}
+                textFieldAppearance={true}
+            />
+        );
+
+        const intervalDisplayValues = [];
         // both ends open
         if (start === undefined && end === undefined) {
-            displayValues.push('All Numbers');
+            intervalDisplayValues.push('All Numbers');
         } else if (start === undefined) {
-            displayValues.push(`≤ ${formatValue(end)}`);
+            intervalDisplayValues.push(`≤ `);
+            intervalDisplayValues.push(endText);
         } else if (end === undefined) {
-            displayValues.push(`> ${formatValue(start)}`);
+            intervalDisplayValues.push(`> `);
+            intervalDisplayValues.push(startText);
         } else if (start === end) {
-            displayValues.push(`${formatValue(start)}`);
+            intervalDisplayValues.push(startEqualsEndText);
         } else if (numericals[0].start === numericals[0].end) {
-            displayValues.push(
-                `${formatValue(start)} ≤ x ≤ ${formatValue(end)}`
-            );
+            intervalDisplayValues.push(startText);
+            intervalDisplayValues.push(` ≤ x ≤ `);
+            intervalDisplayValues.push(endText);
         } else {
-            displayValues.push(
-                `${formatValue(start)} < x ≤ ${formatValue(end)}`
-            );
+            intervalDisplayValues.push(startText);
+            intervalDisplayValues.push(` < x ≤ `);
+            intervalDisplayValues.push(endText);
         }
+
+        displayValues.push(
+            stringOutput
+                ? intervalDisplayValues.join('')
```

Comment:
it seems odd that you either push a string or an array of strings.  is there some flattening later?

Reply:
the array would actually be of jsx elements (e.g. startText which is actually an EditableSpan), so its a different use case

Reply:
i see.  so like a component, you can pass it children that are either string or array of things